scipy/optimize/_qap.py

```diff
@@ -148,39 +149,47 @@ def quadratic_assignment(
 
     Examples
     --------
-
+    >>> import numpy as np
+    >>> from scipy.optimize import quadratic_assignment
     >>> cost = np.array([[0, 80, 150, 170], [80, 0, 130, 100],
     ...         [150, 130, 0, 120], [170, 100, 120, 0]])
     >>> dist = np.array([[0, 5, 2, 7], [0, 0, 3, 8],
     ...         [0, 0, 0, 3], [0, 0, 0, 0]])
-    >>> from scipy.optimize import quadratic_assignment
     >>> res = quadratic_assignment(cost, dist)
     >>> print(res)
      col_ind: array([0, 3, 2, 1])
          nit: 9
        score: 3260
 
-    To demonstrate explicitly how the `score` value
-    :math:`f(P) = trace(A^T PBP^T )` is calculated, one may construct the
-    permutation matrix, and perform the necessary algebra.
+    The see the relationship between the returned ``col_ind`` and ``score``,
+    use ``col_ind`` to form the best permutation matrix found, then evaluate
+    the objective function :math:`f(P) = trace(A^T P B P^T )`.
 
     >>> n = cost.shape[0]
-    >>> P = np.zeros((n, n))
-    >>> P[np.arange(n), res['col_ind']] = 1
+    >>> perm = res['col_ind']
+    >>> P = np.eye(n)[perm]
```

Comment:
It may be more intuitive to express a permutation matrix as a permutation of the identity matrix rather than a zero matrix with ones added in certain locations.